enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/EnterpriseSecurityModule.java

```diff
@@ -258,11 +263,30 @@ private static CacheManager createCacheManager( Config config )
                 {
                     PluginRealm pluginRealm =
                             new PluginRealm( null, plugin, config, securityLog, Clocks.systemClock(), secureHasher );
-                    realms.add( pluginRealm );
+                    availablePluginRealms.add( pluginRealm );
                 }
             }
         }
 
+        List<PluginRealm> realms =
+                availablePluginRealms.stream()
+                        .filter( realm -> configuredRealms.contains( realm.getName() ) )
+                        .collect( Collectors.toList() );
+        boolean missingAuthenticationRealm = pluginAuthenticationEnabled &&
+                !realms.stream().anyMatch( PluginRealm::canAuthenticate );
+        boolean missingAuthorizingRealm = pluginAuthorizationEnabled &&
+                !realms.stream().anyMatch( PluginRealm::canAuthorize );
+
+        if ( missingAuthenticationRealm || missingAuthorizingRealm )
```

Comment:
no, because this only fails if you explicitly put plugin_authentication_enabled=true, but no plugin authentication realm was loaded.